protocol/communities_messenger_signers_test.go

```diff
@@ -732,3 +734,135 @@ func (s *MessengerCommunitiesSignersSuite) TestSyncTokenGatedCommunity() {
+		PrivilegesLevel: token.OwnerLevel,
+	}, nil)
+	s.Require().NoError(err)
+
+	// john adds minted owner token to community
+	err = s.john.AddCommunityToken(community.IDString(), int(chainID), tokenAddress)
+	s.Require().NoError(err)
+
+	// update mock - the signer for the community returned by the contracts should be john
+	s.collectiblesServiceMock.SetSignerPubkeyForCommunity(community.ID(), common.PubkeyToHex(&s.john.identity.PublicKey))
+	s.collectiblesServiceMock.SetMockCollectibleContractData(chainID, tokenAddress,
+		&communitytokens.CollectibleContractData{TotalSupply: &bigint.BigInt{}})
+
+	// alice accepts community update
+	_, err = WaitOnSignaledMessengerResponse(
+		s.alice,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && len(r.Communities()[0].TokenPermissions()) == 1
+		},
+		"no communities",
+	)
+	s.Require().NoError(err)
+
+	grantPermission(&s.Suite, community, s.john, s.alice, protobuf.CommunityMember_ROLE_ADMIN)
+	s.advertiseCommunityTo(s.john, community, s.bob)
+	s.john = nil
+
+	err = tt.RetryWithBackOff(func() error {
+		_, err = johnsOtherDevice.RetrieveAll()
+		if err != nil {
+			return err
+		}
+
+		// Do we have new synced community settings?
+		syncedSettings, err := johnsOtherDevice.communitiesManager.GetCommunitySettingsByID(community.ID())
+		if err != nil || syncedSettings == nil {
+			return fmt.Errorf("community with sync not received %w", err)
+		}
+		return nil
+	})
+
+	request := requests.CreateCommunityTokenPermission{
+		CommunityID: community.ID(),
+		Type:        protobuf.CommunityTokenPermission_BECOME_MEMBER,
+		TokenCriteria: []*protobuf.TokenCriteria{
+			&protobuf.TokenCriteria{
+				Type:              protobuf.CommunityTokenType_ERC20,
+				ContractAddresses: map[uint64]string{testChainID1: "0x123"},
+				Symbol:            "TEST",
+				AmountInWei:       "100000000000000000000",
+				Decimals:          uint64(18),
+			},
+		},
+	}
+
+	response, err := s.alice.CreateCommunityTokenPermission(&request)
+	s.Require().NoError(err)
+	s.Require().Len(response.CommunityChanges, 1)
+	s.Require().Len(response.CommunityChanges[0].TokenPermissionsAdded, 1)
+
+	addedPermission := func() *communities.CommunityTokenPermission {
+		for _, permission := range response.CommunityChanges[0].TokenPermissionsAdded {
+			return permission
+		}
+		return nil
+	}()
+	s.Require().NotNil(addedPermission)
+	// Permission added by event must be in pending state
+	s.Require().Equal(communities.TokenPermissionAdditionPending, addedPermission.State)
+
+	// Retrieve community link & community
+	err = tt.RetryWithBackOff(func() error {
+		response, err = johnsOtherDevice.RetrieveAll()
+		if err != nil {
+			return err
+		}
+		if len(response.Communities()) == 0 {
+			return errors.New("no communities received")
+		}
+		return nil
+	})
+	s.Require().NoError(err)
+
+	s.collectiblesServiceMock.SetSignerPubkeyForCommunity(community.ID(), common.PubkeyToHex(&s.alice.identity.PublicKey))
```

Comment:
This looks wrong. Why alice is the signer now? :thinking:

Reply:
this is a mistake, the other device from John is the one supposed to be set control node. Fixed